worldeditadditions_commands/commands/selectors/srect.lua

```diff
@@ -0,0 +1,51 @@
+-- ███████ ██████  ███████  ██████ ████████
+-- ██      ██   ██ ██      ██         ██
+-- ███████ ██████  █████   ██         ██
+--      ██ ██   ██ ██      ██         ██
+-- ███████ ██   ██ ███████  ██████    ██
+-- lua parse_params_srect("10")
+-- local
+function parse_params_srect(params_text)
+  local find, _, sn1, ax1, sn2, ax2, len = params_text:find("([+-]?)([xyz]?)%s*([+-]?)([xyz]?)%s*(%d*)")
+
+  -- If ax1 is nil set to player facing dir
+  if ax1 == "" then ax1 = "get"
+  else ax1 = {tonumber(sn1..1),string.lower(ax1)}
+	end
+  -- If ax2 is nil set to +y
+  if ax2 == "" then ax2 = "y" end
+  ax2 = {tonumber(sn2..1),string.lower(ax2)}
+
+  len = tonumber(len)
+  if len == nil then
+    return false, "No length specified."
+  end
+
+  return true, ax1, ax2, len
+end
+worldedit.register_command("srect", {
+	params = "[<axis1> [<axis2>]] <length>",
+	description = "Set WorldEdit region position 2 at a set distance along 2 axes.",
+	privs = {worldedit=true},
+  require_pos = 1,
+  parse = function(params_text)
+		local values = {parse_params_srect(params_text)}
```

Comment:
Not sure this really needs to be a separate function here. There's only this command using it, so I don't think it matters.

Reply:
keeping it as a separate function allows me to debug it better. I may merge it into the command later though.

Reply:
Ah, ok. No problem! That makes sense.